src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -163,4 +171,17 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
             s"from available source columns '${dfColumnNames.mkString(",")}'")
     }
   }
+
+  private def internalColumns(df: DataFrame, indexConfigColumns: Seq[String]): Seq[String] = {
+    val partitionSchema = df.queryExecution.optimizedPlan.collect {
+      case LogicalRelation(HadoopFsRelation(_, pSchema, _, _, _, _), _, _, _) => pSchema
+    }
+
+    val spark = df.sparkSession
+    partitionSchema.head
+      .map(_.name)
+      .filter(ResolverUtils.resolve(spark, _, indexConfigColumns).isEmpty)
+  }
+
+  private val fileName = udf((fullFilePath: String) => FilenameUtils.getBaseName(fullFilePath))
```

Comment:
can you add a type since this is `private val`? Also, should we change the variable name since this is a function?

Reply:
Done.